snapyr/src/main/java/com/snapyr/sdk/notifications/SnapyrNotificationListener.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014 Segment.io, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.snapyr.sdk.notifications;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+
+import androidx.core.app.NotificationManagerCompat;
+
+import com.snapyr.sdk.Snapyr;
+import com.snapyr.sdk.internal.Utils;
+
+
+/**
+ * SnapyrNotificationListener
+ * Listens for broadcast events for the action `@package_name.TRACK_BROADCAST` and fires off
+ * a track event with the payload from the intent. This is currently used by action buttons to
+ * trigger track notifications instead of creating a new activity. If the action button specifies
+ * a deeplink then the listener will attempt to transition to the URI provided.
+ */
+public class SnapyrNotificationListener extends BroadcastReceiver {
+    private static final String TAG = "SnapyrNotificationListener";
+
+
+
+//    @Override
+//    public void onCreate(Bundle bundle){
+//        super.onCreate(bundle);
+//
+//        Intent startIntent = new Intent();
+//        startIntent.setAction("android.intent.action.MAIN");
+//        startIntent.addCategory("android.intent.category.LAUNCHER");
+//        startIntent.setData(Uri.parse("snapyrsample://test"));
+//        startIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+//        this.startActivity(startIntent);
+//    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        String actionId =  intent.getStringExtra(SnapyrNotificationHandler.ACTION_ID_KEY);
```

Comment:
We're using the action's `id` field (i.e. Snapyr-generated UUID, rather than user-defined "actionId") in iOS land, and that's what we figured on connecting in the back end. 

My bad for not mentioning that earlier. Now that I think of it again, I think the user-defined "actionId" was thought of as an iOS-only thing, but we're not using that anymore. I think we can maybe remove it altogether from push templates (@tehpsalmist , does that sound right to you?)